test/arguments.ts

```diff
@@ -268,3 +268,19 @@ test('throws if the `searchParams` value is invalid', async t => {
 		message: 'The `searchParams` value \'\' must be a string, number, boolean or null'
 	});
 });
+
+test('`userData` option is not enumerable', withServer, async (t, server, got) => {
+	server.get('/', echoUrl);
+
+	await got({
+		userData: 123,
+		hooks: {
+			beforeRequest: [
+				options => {
+					t.is(options.userData, 123);
+					t.false({}.propertyIsEnumerable.call(options, 'userData'));
+				}
+			]
+		}
+	});
+});
```

Comment:
I think we need some more tests. For example, ensure that `userData` is correctly preserved when extending the Got instance, and that it's available in hooks.